### PR TITLE
CT-1599 record searches and clicks

### PR DESCRIPTION
**Pull request by stephenrichards**

### CODE REVIEW

**Review by aliuk2012 — COMMENTED** [review]
Only reviewed from Frontend POV

File: app/models/search_query.rb [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+      )
+    end
+  end
+
+
+  def self.update_for_click(params)
+    record = SearchQuery.find_by(query_hash: params[:hash])
+    unless record.nil?
+      record.num_clicks +=1
+      if record.highest_position.nil? || record.highest_position > params[:pos].to_i
```
Comment (misaka) [inline app/models/search_query.rb]: Is this right that we only want to update the `highest_position` if `params[:pos]` is _lower_ than it? I expected it to be the inverse, if `params[:pos]` was higher, and the test below appears to test when `highest_position = 3` and `params[:pos] = 3` so it's not quite clear.

File: spec/controllers/cases_controller/show_spec.rb [hunk 2]
```diff
@@ -374,6 +374,31 @@
...
+      context 'has a hash parameter' do
+        it 'records search query to record the click' do
+          params = ActionController::Parameters.new(id: kase.id.to_s, hash: "XYZ", controller: "cases", action: "show")
+          expect(SearchQuery).to receive(:update_for_click).with(params)
+          get :show, params: params.to_unsafe_hash
+        end
+      end
+
+      context 'does not have a hash parameter' do
+        it 'records search query to record the click' do
```
Comment (misaka) [inline spec/controllers/cases_controller/show_spec.rb]: I think this should read:

```
it 'does not record search query' do
```

File: spec/helpers/cases_helper_spec.rb [hunk 3]
```diff
@@ -160,5 +160,31 @@
...
+    context 'no query hash instance variable' do
+      it 'shows link without hash and position parameters' do
+        expected_link = "<a href=\"/cases/25\">180425001</a>"
+        expect(case_link_with_hash(kase, :number, nil, 3, 44)).to eq expected_link
+      end
+    end
+
+    context 'query hash instance variable exists' do
+      context 'page parameters exists' do
+        it 'shows link without hash and position parameters' do
```
Comment (misaka) [inline spec/helpers/cases_helper_spec.rb]: Spec description is wrong.

File: spec/helpers/cases_helper_spec.rb [hunk 4]
```diff
@@ -160,5 +160,31 @@
...
+          expected_link = "<a href=\"/cases/25?hash=XYZ&amp;pos=35\">180425001</a>"
+          expect(case_link_with_hash(kase, :number, 'XYZ', 2, 14)).to eq expected_link
+        end
+      end
 
+      context 'page number does not exist' do
+        it 'shows link without hash and position parameters' do
```
Comment (misaka) [inline spec/helpers/cases_helper_spec.rb]: Spec description is wrong.

**Pull request closed without merging**